R/sircovid.R

```diff
@@ -1,7 +1,127 @@
 ##' Create a model
+##' 
 ##' @title Create a model
+##' 
+##' @param params List of parameters of the model. Get
+##' with \code{generate_parameters}
+##' 
+##' @returns model object for use with \code{particle_filter()}
+##' 
 ##' @export
-##' @param params List of parameters of the model
 sircovid <- function(params) {
   basic(user = params)
 }
+
+##' Create a model, and fit with the particle filter
+##' 
+##' @title Run particle filter
+##' 
+##' @param case_data_file CSV file containing COVID-19 cases, from
+##'   the ECDC website 
+##' 
+##' @param admissions_data_file Admission totals, from ncov-outputs
+##' 
+##' @param itu_colname Name of the column in the time series file
+##'   to be used for fitting ITU cases
+##' 
+##' @param itu_ignore_dates List of dates to discard from the
+##'   data due to error or otherwise
+##'   
+##' @param first_data_date Date to start data fit from
+##' 
+##' @param model_start_date Date to run model simulations from
+##' 
+##' @param country_name Country name in case reports
+##' 
+##' @param case_file_day_shift Number of days to subtract from
+##'  dateRep in case file by to match with date in 
+##'  \code{time_series_file}
+##'  
+##' @param model_params Parameters to run model simulations,
+##'   generated by \code{generate_parameters()}
+##' 
+##' @param obs_params List of parameters used for comparing
+##'   model to data in the particle filter
+##'   
+##' @param n_particles Number of particles  
+##' 
+##' @returns Results from particle filter
+##' 
+##' @export
+run_particle_filter <- function(case_data_file,
```

Comment:
case_data_file is actually deaths? i think it is a bit confusing. I think at the moment it is not very generic, i.e. we will have soon to add the bed time series also. I think though we should stick to this at the moment to get this working. We can think of a more generic function later, and do a bit of brain storm. It would be good to have a bit of a brainstorm too as we come from different background and we want to encompass the statistic/mathematic logic but with the architecture/computer science robustness and flexibility

Reply:
I don't think it's too bad for a start, as it takes two files which just need to have date columns, one deaths and one itu (with arbitrary name). If this is expanded later to more files and column names, probably a more general `input_files`, `date_columns` and `output_columns` could be used, but don't know whether this is worth it now. What do you think?

Will change case to deaths if that makes it clearer - named cases because the downloaded file is called case data, though it also contains deaths (which I realise is what is actually being used)